src/cargo/ops/cargo_rustc/custom_build.rs

```diff
@@ -364,6 +391,13 @@ impl BuildOutput {
                 _ => bail!("Wrong output in {}: `{}`", whence, line),
             };
 
+            let path = |val: &str| {
```

Comment:
Could we, instead of post-processing after reading, do some pre-processing before writing, so as not to store absolute paths at the moment of generation at all?

Reply:
Hm yeah I wasn't sure what was the best option here, we don't have a serialized format for Cargo's parsed data structures for the output file so it'd have to be added, it just wasn't clear to me how much of a benefit we'd get from it.

Do you think that the preprocessed version would be less buggy though?

Reply:
I don't think there are any tangible benefits one way or the other, but not having absolute paths feels slightly cleaner. But I am fine either way!